src/Microsoft.DncEng.PatGeneratorTool/Program.cs

```diff
@@ -0,0 +1,169 @@
+using Azure.Core;
+using Azure.Identity;
+using Microsoft.DncEng.PatGenerator;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using Microsoft.VisualStudio.Services.Client;
+using Microsoft.VisualStudio.Services.Common;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Threading.Tasks;
+
+namespace Microsoft.DncEng.PatGeneratorTool
+{
+    class Program
+    {
+        private const int DefaultExpirationInDays = 30;
+
+        static int Main(string[] args)
+        {
+            var possibleEnumScopes = string.Join(", ", Enum.GetNames(typeof(AzureDevOpsPATScopes)));
+            var cmd = new RootCommand
+            {
+                new Option<List<AzureDevOpsPATScopes>>("--scopes",
+                    parseArgument: arg => ParsePATScopes(arg),
+                    description: $"PAT scopes. Valid values are: {possibleEnumScopes}.")
```

Comment:
[nit] I wonder if it makes sense to mention that the permissions are ordered from "least access to most access" in each category? It might not be clear to users that `build` means only `build_read` and not "everything in the build section"